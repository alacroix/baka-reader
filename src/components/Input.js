// @flow

import React from 'react';
import { StyleSheet, TextInput } from 'react-native';
import appStyle from 'bakareader/src/appStyle';

const styles = StyleSheet.create({
  input: {
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
    paddingHorizontal: appStyle.grid.x1,
  },
});

type PropsType = {
  type: 'url',
};

function Input(props: PropsType) {
  switch (props.type) {
    case 'url':
      return <TextInput autoCorrect={false} keyboardType="url" autoCapitalize="none" style={styles.input} {...props} />;
    default:
      return <TextInput />;
  }
}

export default Input;
