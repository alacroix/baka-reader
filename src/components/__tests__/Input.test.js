import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Input from '../Input';

describe('Input', () => {
  it('should render an URL formatted input ', () => {
    const tree = renderer.create(
      <Input type="url" />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render an default input if no type is provided', () => {
    const tree = renderer.create(
      <Input />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
