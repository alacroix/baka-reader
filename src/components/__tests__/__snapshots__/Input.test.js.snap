// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render an URL formatted input  1`] = `
<TextInput
  autoCapitalize="none"
  autoCorrect={false}
  keyboardType="url"
  style={
    Object {
      "borderColor": "gray",
      "borderWidth": 1,
      "height": 40,
      "paddingHorizontal": 8,
    }
  }
  type="url"
/>
`;

exports[`Input should render an default input if no type is provided 1`] = `<TextInput />`;
